transport/http.go

```diff
@@ -306,7 +313,10 @@ func (t *HTTPTransport) SetServerURL(u ...*url.URL) error {
 	t.intakeURLs = intakeURLs
 	t.configURLs = configURLs
 	t.profileURLs = profileURLs
-	t.urlIndex = 0
+	atomic.StoreInt32(&t.urlIndex, 0)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	t.refreshRemoteVersion(ctx)
```

Comment:
Now that I think about it some more, I think this side-effect of SetServerURL could be a little surprising. It would be surprising to me that setting URLs might block for up to 5 seconds.

Do you think it's important to eagerly fetch the version? Maybe just fetch it on the first call to MajorServerVersion?